### DISCUSSION TITLE
Changing baud rate on instrumented builds

### DISCUSSION THREAD

**Discussion by cmidgley** [Moddable-OpenSource/moddable#1409] [Ideas]
Currently, on ESP32 at least, instrumented builds have the serial port hard-coded to 115,200 baud and it can't be changed without modifying the `/build/devices/...` source code for the device.

I've only looked at ESP32, but here one example of [source code](https://github.com/Moddable-OpenSource/moddable/blob/a9e80d0c19703fc26e26084d7b36976d95e8211a/build/devices/esp32/xsProj-esp32/main/main.c#L240C1-L244C7) that hard-codes the baud rate to 115,200 for everything except a debug build (which uses the `DEBUGGER_SPEED`  macro).  This is typical across a wide range of board implementations.

The problem is that if I do an instrumented build, it forces me to set the overall baud rate to 115,200 (including upload if I want to use `mcconfig` to build and deploy) in order to see the instrumented output.  If you change the ESP32 to use a different default baud rate (via `sdkconfig.defaults` with `CONFIG_ESP_CONSOLE_UART_BAUDRATE=460800` and `CONFIG_ESP_CONSOLE_UART_CUSTOM=y`) then the ESP-IDF diagnostics output is at 460,800, but after that the board module switches it to 115,200.  The only solution I've found is to edit the board-specific source mentioned above to 460,800, and then it works perfectly.

Since both debug and instrumented builds take over the serial port, I was thinking it would be logical for the `DEBUGGER_SPEED` to be set for both types of builds (leaving release alone), or perhaps introduce a new macro like `INSTRUMENTED_SPEED`.

I didn't want to proceed with a PR for this without a discussion in advance.  Interested in your perspective on if I am approaching this in the correct way, if a source change is desirable, and your thoughts on how you might like to see it done (what macro to use, and best way to compile-time detect an instrumented build).

**Comment by phoddie** [comment]
Yes, the Moddable SDK default matches the default of the ESP-IDF consolel. Maybe the easiest solution is just to use `CONFIG_ESP_CONSOLE_UART_BAUDRATE` in an instrumented build on ESP-IDF. That would be just a single setting, so there's nothing to get out of sync.

**Comment by cmidgley** [comment]
Implemented as PR #1413
